ambry-rest/src/main/java/com.github.ambry.rest/ConnectionStatsHandler.java

```diff
@@ -0,0 +1,70 @@
+/**
+ * Copyright 2015 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.rest;
+
+import io.netty.channel.ChannelDuplexHandler;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.channel.ChannelPromise;
+import java.net.SocketAddress;
+import java.util.concurrent.atomic.AtomicLong;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * Handler that tracks connection establishment statistics.
+ */
+class ConnectionStatsHandler extends ChannelDuplexHandler {
```

Comment:
http://netty.io/4.0/api/io/netty/channel/ChannelHandler.Sharable.html

You use the annotation when you don't want a new instance of the class created for every connection (which is the default). In essence, you are saying that the same object can be shared across multiple connections without creating a race condition (same as what you are current code is doing now).